browser/ipfs/ipfs_tab_helper_browsertest.cc

```diff
@@ -849,4 +850,95 @@ IN_PROC_BROWSER_TEST_F(IpfsTabHelperBrowserTest, IPFSPromoInfobar_NowShown) {
+  ASSERT_TRUE(helper);
+  auto* prefs =
+      user_prefs::UserPrefs::Get(active_contents()->GetBrowserContext());
+  prefs->SetInteger(
+      kIPFSResolveMethod,
+      static_cast<int>(ipfs::IPFSResolveMethodTypes::IPFS_GATEWAY));
+  prefs->SetBoolean(kIPFSAutoRedirectToConfiguredGateway, true);
+  GURL gateway_url = embedded_test_server()->GetURL("navigate.to", "/");
+  prefs->SetString(kIPFSPublicGatewayAddress, gateway_url.spec());
+
+  GURL gateway = ipfs::GetConfiguredBaseGateway(prefs, chrome::GetChannel());
+
+  const GURL test_url = embedded_test_server()->GetURL(
+      "drweb.link",
+      "/ipns/k2k4r8ni09jro03sto91pyi070ww4x63iwub4x3sc13qn5pwkjxhfdt4");
+
+  auto find_infobar =
+      [](infobars::ContentInfoBarManager* content_infobar_manager)
+      -> infobars::InfoBar* {
+    for (size_t i = 0; i < content_infobar_manager->infobar_count(); i++) {
+      auto* infobar = content_infobar_manager->infobar_at(i);
+      if (infobar->delegate()->GetIdentifier() ==
+          BraveConfirmInfoBarDelegate::BRAVE_IPFS_FALLBACK_INFOBAR_DELEGATE) {
+        return infobar;
+      }
+    }
+    return nullptr;
+  };
+
+  //  Disable IPFS Companion
+  prefs->SetBoolean(kIPFSCompanionEnabled, false);
+  {
+    ui_test_utils::NavigateToURL(browser(), test_url);
+    WaitForLoadStopWithoutSuccessCheck(active_contents());
+    // Get last shown infobar
+    auto* infobar = find_infobar(
+        infobars::ContentInfoBarManager::FromWebContents(active_contents()));
+    //  IPFS Fallback Infobar should not be shown
+    ASSERT_FALSE(infobar);
+  }
+
+  SetHttpStatusCode(net::HTTP_INTERNAL_SERVER_ERROR);
+
+  {
+    ui_test_utils::NavigateToURL(browser(), test_url);
+    WaitForLoadStopWithoutSuccessCheck(active_contents());
+    // Get last shown infobar
+    auto* infobar = find_infobar(
+        infobars::ContentInfoBarManager::FromWebContents(active_contents()));
+    //  IPFS Fallback Infobar should be shown
+    ASSERT_TRUE(infobar);
+    static_cast<BraveConfirmInfoBar*>(infobar)->GetDelegate()->Accept();
+    WaitForLoadStopWithoutSuccessCheck(active_contents());
+    //  Redirected to original address
+    EXPECT_EQ(active_contents()->GetVisibleURL(), test_url);
+  }
+
+  {
+    ui_test_utils::NavigateToURL(browser(), test_url);
+    WaitForLoadStopWithoutSuccessCheck(active_contents());
+    auto ipfs_address = active_contents()->GetVisibleURL();
+    // Get last shown infobar
+    auto* infobar = find_infobar(
+        infobars::ContentInfoBarManager::FromWebContents(active_contents()));
+    //  IPFS Fallback Infobar should be shown
+    ASSERT_TRUE(infobar);
+    static_cast<BraveConfirmInfoBar*>(infobar)->GetDelegate()->Cancel();
+    WaitForLoadStopWithoutSuccessCheck(active_contents());
+    //  Stayed on the same address
+    EXPECT_EQ(active_contents()->GetVisibleURL(), ipfs_address);
+  }
+
+  //  Enable the IPFS companion
+  prefs->SetBoolean(kIPFSCompanionEnabled, true);
```

Comment:
Please add a case when infobar is not shown when navigating to some non-ipfs related url, like brave.com and getting a error

Reply:
done